src/resources.ts

```diff
@@ -16,27 +16,31 @@ import {
 
 function useAuthenticatedHttpResource<ResourceT>(
   initialState: ResourceT,
-  fetcher: (options: AuthenticatedHttpOptions) => Promise<ResourceT>,
+  fetcher: (options: AuthenticatedHttpOptions) => Promise<ResourceT>
 ) {
   const [resource, setResource] = useState<ResourceT>(initialState);
   const [loading, setLoading] = useState(false);
   const { token } = useAuthentication();
 
-  useEffect(() => {
-    const cancelToken = http.CancelToken.source();
-    const load = async () => {
+  const loadResource = useCallback(
```

Comment:
Why do we need the `useCallback`?

Reply:
If you mean as in _why do we separate this_: Then because we want to expose this function as well (for reloading)

If you mean as in _why use the hook_ then because the effect below is actually dependant on both the token and the fetcher. If those change, the scope of the effect is out of sync. If we want to separate the `loadResource`, we can make that dependent on `token` and `fetcher`, then make the effect dependant on `loadResource`. This way the effect's dependencies are correctly represented, meaning if they change the scope changes.

I recommend reading specifically this section in depth, this covers this scenario well and gives good tips: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

When i started working with hooks i kind of disregarded the exhaustive-deps rule, now i tend to follow it almost as law (there are exceptions ofc, but they're extremely rare), because this stuff kept causing subtle bugs that were hard to track down.

There is actually one more reason: I am using `loadResource` in useEffects in other places, meaning it should not change every render or we'll go into an infinite loop. Generally when returning functions from a hook, I find it's very often a good idea to wrap them with `useCallback`.

Reply:
Thanks for explaining!

I understand why we separate this, why to ensure dependencies are exhaustive, and that changing `loadResource` in every render would trigger an infinite loop, will look more into it to understand why we couldn't make do with just having `token` in dependencies everywhere.

Reply:
We could also add that. I think it's easier to grok (especially for automatic checks with eslint) if you only expose the direct dependencies.